### PR TITLE
fix: snackbar label width miscomputed when image is set

### LINKED ISSUE #117: bug: snackbar label
**Describe the bug**
In some precise circumstances, the message of a snackbar can be truncated (and ends with '...' characters)

**Steps to reproduce**

Display a snackbar with the following specs: 
- an image
- whatever title
- the following message : 'Votre changement a été pris en compte' (or any massage with the same number of characters)

You can see on the bug in the screenshot below

But if the message is a little longer (for example, "Votre changement a été correctement pris en compte"), it works as expected

**Expected behavior**

Th message should always go on a new line if it is too long, whatever its length

**Screenshots**

image

**Device (please complete the following information):**
- Model: iPhone 8
- OS: iOS
- Version 16.1

### PR DESCRIPTION
**Pull request by daniel-dumortier**
## Changes description
In some circumstances, then label of a snackbar with an image can be truncated with '...' characters at the end.
It depends on the length of the message, that should be a specific lenght (not too short, not too long), that's why it has not been detected earlier

I took advantage of this PR to add completion on `display()` and `dismiss()` methods of `VitaminSnackbar` as requested by a team that uses the snackbar without the `UIViewController` extension.

## Context
Closes #117

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [ ] I have reviewed the submitted code.
- [x] I have tested on an iPhone device/simulator.
- [x] I have tested on an iPad device/simulator.
- [ ] If it includes design changes, please ask for a review `design-system-core-team-design` GitHub team.

## Does this introduce a breaking change?

- No

#### iPhone

Simulator Screen Shot - iPhone 8 - 2023-02-12 at 21 19 05

### CODE REVIEW

File: Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift [hunk 1]
```diff
@@ -286,9 +286,10 @@ extension VitaminSnackbar {
     }
 
     private func updateMessageLabelPreferredMaxWidth() {
-        if image != nil {
-            let margins = rightMargin + imageWidthAndHeight + imageLeftAndRightMargin
+        guard image == nil else {
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift:289]: I'm not sure about the usage of `guard`. It's maybe better to keep `guard` only for dependency checks and not for the condition on a feature 🤔 But it's an open debate 😅

**Review by florentlotthepro — APPROVED** [review]
You update the documentation, you can maybe also update `VitaminSnackBar+Dismiss.swift` -> `VitaminSnackbar+Dismiss.swift`

File: Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift (hunk 1 above)
Comment (daniel-dumortier) [reply Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift:289]: While using an ìf`, I forgot to write my `return` statement (or `else`), leading to the issue.
And that would have not been possible with this `guard` syntax, that is why I switched.
But ok to switch back to `if`;)

Comment (florentlotthepro) [reply Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift]: Yes, I didn't think about this use case 🤔

**Review by florentlotthepro — APPROVED** [review]

**Comment by daniel-dumortier** [comment]
Sorry, I spotted two errors in documentation, and fixed them. That should be the last one before merge ;)

**Review by florentlotthepro — APPROVED** [review]

**Pull request merged**